browser/brave_screenshots/screenshots_utils.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/brave_screenshots/screenshots_utils.h"
+
+#include "chrome/browser/image_editor/screenshot_flow.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_window.h"
+#include "chrome/browser/ui/browser_window/public/browser_window_interface.h"
+#include "chrome/browser/ui/tabs/public/tab_interface.h"
+#include "ui/base/clipboard/scoped_clipboard_writer.h"
+
+namespace brave_screenshots {
+namespace utils {
+void NotifyUserOfScreenshot(const image_editor::ScreenshotCaptureResult& result,
+                            base::WeakPtr<content::WebContents> web_contents) {
+  if (!web_contents || result.image.IsEmpty()) {
+    return;
+  }
+
+  // Copy the image to the user's clipboard
+  ui::ScopedClipboardWriter(ui::ClipboardBuffer::kCopyPaste)
+      .WriteImage(*result.image.ToSkBitmap());
+
+  // Notify the user that the screenshot has been taken
+  tabs::TabInterface* tab =
+      tabs::TabInterface::GetFromContents(web_contents.get());
+
+  // Leverage the screenshot bubble to show the user the screenshot
+  tab->GetBrowserWindowInterface()
+      ->GetBrowserForMigrationOnly()
```

Comment:
Is there something we can use instead of this `MigrationOnly` method? The comment on it seems to indicate we shouldn't be using it:
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/browser_window/public/browser_window_interface.h;l=201;bpv=0;bpt=1

Reply:
I can pass a pointer to the browser window along to this method, but my concern is that the pointer may or may not refer to the actual host window when the screenshot operation is complete. As such, it would be nice if we had a route to get the browser window from the WebContents, but I don't see a preferred solution for that anywhere.

Reply:
There's also `chrome::FindBrowserWithTab` which probably does what you want (though the docs discourage its use too).

cc @goodov

Reply:
Yeah, I am avoiding global routes (per the docs). I think I have a suitable approach in the latest changes, but we'll see following further review :)